### DISCUSSION TITLE
Foundry or Hardhat

### DISCUSSION THREAD

**Discussion by mumernisar** [Cyfrin/foundry-full-course-cu#2516] [General]
Hey everyone,

I'm new to this career and have some experience with the MERN stack, as well as some freelancing work. Now I'm looking to transition to web3, and I'm a bit confused because of my background in JavaScript. I've started learning the fundamentals of Solidity, but I'm not sure where to go next. Should I focus on Hardhat, since it's in JS and I'm already familiar with it? I've heard that it's relatively easy and can make me more marketable for jobs. However, You people are the experts and might have more insight. I'd really appreciate some suggestions or a roadmap/resource that could help me become a better engineer/developer.

Thanks for taking the time to read this. 🥲💖

**Comment by EngrPips** [comment]
Hello @mumernisar, I will say you can start with `Hardhat` if that is what you feel more comfortable with, but you should understand that Cyfrin does not teach Hardhat for now but teaches Foundry instead. So you might need to learn Hardhat separately while you take Cyfrin courses that, in reality, you learning both Hardhat and Foundry simultaneously. Talking about the roadmap all the courses on Cyfrin Updraft are everything you need to have a solid foundation.

**Comment by mumernisar** [comment]
Thank you @EngrPips. Really appreciate your response. But as I have also taken a solidity course from Cyfrin which I think is only the basics which they advance later in the foundry or security section (I guess). So if were to start Hardhat I will also have to learn more about solidity through their documentation or some other resource(which there arent many ig). So should I really go for hardhat now?
Or could I learn foundry now and then I will have ease in learning hardhat while also having deep solidity knowledge?
I apologize if my questions don't make much sense.

**Reply by EngrPips** [reply]
I will advise learn Foundry now, and that can be easily done with Cyfrin course, and if the need arises you will learn Hardhat.

**Reply by mumernisar** [reply]
Thanks for the help ..!! 💖
I think I will do so..!

**Reply by EngrPips** [reply]
You are welcome, Friend. I wish you the best on your journey!

**Comment by Aashim-Limbu** [comment]
yeah i'm also having same dilemma weather to go with hardhat or solidity.Isn 't there any opensource course like cyfrin for hardhat ?

**Reply by EngrPips** [reply]
I haven't searched for one, but I think there should be. Patrick has a Hardhat course on Youtube, but it is outdated now.

**Reply by Aashim-Limbu** [reply]
Thank you 👍.Anyway i decided to go with foundry 👍.

**Reply by EngrPips** [reply]
Not a bad choice I will say.
